client/src/app/core/core-services/http.service.ts

```diff
@@ -273,4 +273,27 @@ export class HttpService {
     public async delete<T>(path: string, data?: any, queryParams?: QueryParams, header?: HttpHeaders): Promise<T> {
         return await this.send<T>(path, HTTPMethod.DELETE, data, queryParams, header);
     }
+
+    /**
+     * Retrieves a binary file from the url and returns a base64 value
+     *
+     * @param url file url
+     * @returns a promise with a base64 string
+     */
+    public async convertUrlToBase64(url: string): Promise<string> {
+        return new Promise<string>((resolve, reject) => {
+            const headers = new HttpHeaders();
+            this.get<Blob>(url, {}, {}, headers, 'blob').then(file => {
```

Comment:
uase `await` here by declaring the anonymous function `async`.

Reply:
i did just move it from pdf service to the http service lol

Reply:
Yes, a goood time to clean up ;)